src/main/java/org/breedinginsight/services/OntologyService.java

```diff
@@ -234,4 +241,86 @@ public void revokeOntology(@NotNull UUID programId, @NotNull UUID sharedProgramI
         // Remove record from db
         programOntologyDAO.revokeSharedOntology(sharedOntology);
     }
+
+    public SubscribedProgram subscribeOntology(UUID programId, UUID sharingProgramId) throws DoesNotExistException, UnprocessableEntityException {
+        // Check that program exists
+        Program program = getProgram(programId);
+
+        // Check that shared program exists
+        Optional<ProgramSharedOntologyEntity> optionalSharedOntology = programOntologyDAO.getSharedOntologyById(sharingProgramId, programId);
+        if (optionalSharedOntology.isEmpty()) {
+            throw new DoesNotExistException("Shared ontology between specified programs was not found.");
```

Comment:
Thoughts on this throwing more of an illegal state type of exception?

Reply:
Let's talk more about this over a slack huddle if that works for you? Did you have a specific status in mind? The only other one I can think of is `UnprocessableEntityException` but the `DoesNotExistException` seems to fit better since the user is requesting a specific resource and it doesn't exist. 

The other `UnprocessableEntityException` errors are thrown when the shared record is found, but there is something else that is preventing the action.

Reply:
Talked with Chris, and I'm fine for it to stay as is